DittoHeartbeat/src/main/java/live.ditto.dittoheartbeat/Heartbeat.kt

```diff
@@ -0,0 +1,135 @@
+    val handler = Handler(Looper.getMainLooper())
+
+    val runnable = object : Runnable {
+        override fun run() {
+
+            val timestamp = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ", Locale.getDefault()).format(Date())
+//            val deviceId = config.metaData["deviceId"] as? String ?: ""
+//            val locationId = config.metaData["locationId"] as? String ?: ""
+
+            val info = HeartbeatInfo(
+                id = createCompositeId(config.id, ditto),
+                lastUpdated = timestamp,
+                presence = observePeers(ditto),
+            )
+
+            addToCollection(info, config, ditto)
+            onDataLog(info)
+            handler.postDelayed(this, config.interval) // Repeat every config.interval milliseconds
+        }
+    }
+
+    handler.postDelayed(runnable, 0) // Initial delay
+
+    return handler
+}
+
+fun createCompositeId(configId: Map<String, String>, ditto: Ditto): Map<String, String> {
+    val compositeId: MutableMap<String, String> = configId.toMutableMap()
+    val presenceGraph = ditto.presence.graph
+    compositeId["dittoPeerKey"] = byteArrayToHash(presenceGraph.localPeer.peerKey)
+    return compositeId
+}
+
+fun byteArrayToHash(byteArray: ByteArray): String {
+    val md5 = MessageDigest.getInstance("md5")
+
+    return BigInteger(1, md5.digest(byteArray))
+        .toString(16)
+        .padStart(32, '0')
+}
+
+fun observePeers(ditto: Ditto): Presence? {
+    val presenceGraph = ditto.presence.graph
+    val totalPeers = presenceGraph.remotePeers.size
+    val connectionsList = presenceGraph.remotePeers
+    presence = Presence(totalPeers, connectionsList)
+
+    return presence
+}
+
+fun addToCollection(info: HeartbeatInfo, config: HeartbeatConfig, ditto: Ditto) {
+
+    val doc = mapOf(
+        "_id" to info.id,
+        "interval" to "${config.interval / 1000} sec",
+        "totalPeers" to (info.presence?.totalPeers ?: 0),
+        "lastUpdated" to info.lastUpdated,
+        "presence" to getConnections(info.presence)
+    )
+
+    ditto.store.collection(config.collectionName).upsert(doc)
+}
+
+fun getConnections(presence: Presence?): Map<String, Any> {
+
+    val connectionsMap: MutableMap<String, Any> = mutableMapOf()
+
+    presence?.peers?.forEach { connection ->
+        val connectionsTypeMap = getConnectionTypeCount(connection = connection)
+
+        val connectionMap: Map<String, Any?> = mapOf(
+            "deviceName" to connection.deviceName,
+            "devicePeerKey" to byteArrayToHash(connection.peerKey),
+            "isConnectedToDittoCloud" to connection.isConnectedToDittoCloud,
+//            "totalPeers" to connection.connections.size,
+            "bluetooth" to connectionsTypeMap["bt"],
+            "p2pWifi" to connectionsTypeMap["p2pWifi"],
+            "lan" to connectionsTypeMap["lan"],
+            )
+
+        connectionsMap["peer ${connectionsMap.size + 1}"] = connectionMap
```

Comment:
Please use the peerKey as the key here because otherwise it is not going to be consistent across updates